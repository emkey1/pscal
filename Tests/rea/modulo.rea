int r = 10 % 3;
