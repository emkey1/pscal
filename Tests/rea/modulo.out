{

  "node_type": "PROGRAM",
  "var_type_annotated": "VOID",
  "main_block": 
  {

    "node_type": "BLOCK",
    "var_type_annotated": "VOID",
    "is_global_scope": false,
    "declarations": 
    {

      "node_type": "COMPOUND",
      "var_type_annotated": "VOID",
      "children": [
        {

          "node_type": "VAR_DECL",
          "var_type_annotated": "INT64",
          "left": 
          {

            "node_type": "BINARY_OP",
            "token": {
                "type": "MOD",
                "value": "%"
            },
            "var_type_annotated": "INT64",
            "left": 
            {

              "node_type": "NUMBER",
              "token": {
                  "type": "INTEGER_CONST",
                  "value": "10"
              },
              "var_type_annotated": "INT64",
              "i_val": 0
            }
,
            "right": 
            {

              "node_type": "NUMBER",
              "token": {
                  "type": "INTEGER_CONST",
                  "value": "3"
              },
              "var_type_annotated": "INT64",
              "i_val": 0
            }

          }
,
          "right": 
          {

            "node_type": "TYPE_IDENTIFIER",
            "token": {
                "type": "IDENTIFIER",
                "value": "int"
            },
            "var_type_annotated": "INT64"
          }
,
          "children": [
            {

              "node_type": "VARIABLE",
              "token": {
                  "type": "IDENTIFIER",
                  "value": "r"
              },
              "var_type_annotated": "INT64"
            }
          ]
        }
      ]
    }
,
    "body": 
    {

      "node_type": "COMPOUND",
      "var_type_annotated": "VOID"
    }

  }

}

